test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Formatter/Deserialization/ODataResourceDeserializerTests.cs

```diff
@@ -1249,16 +1252,69 @@ public void ApplyNestedProperty_UsesThePropertyAlias_ForResourceWrapper()
                 new ODataNestedResourceInfoWrapper(new ODataNestedResourceInfo { Name = "Customer" });
             resourceInfoWrapper.NestedItems.Add(new ODataResourceWrapper(resource));
 
-            ODataDeserializerContext context = new ODataDeserializerContext { Model = model.Model };
+            IEdmEntityTypeReference customerTypeReference = model.Model.GetEdmTypeReference(typeof(Customer)).AsEntity();
+
+            IEdmEntityType entityType1 = customerTypeReference.EntityDefinition();
+            EdmEntityContainer container = new EdmEntityContainer("NS", "Container");
+            IEdmNavigationSource navigationSource = new EdmEntitySet(container, "EntitySet", entityType1);
+
+            var keys = new[] { new KeyValuePair<string, object>("ID", 42) };
+
+            ODataDeserializerContext readContext = new ODataDeserializerContext()
+            {
+                Model = model.Model
+            };
+
 
             // Act
             new ODataResourceDeserializer(_deserializerProvider)
-                .ApplyNestedProperty(order, resourceInfoWrapper, model.Order.AsReference(), context);
+                .ApplyNestedProperty(order, resourceInfoWrapper, model.Order.AsReference(), readContext);
 
             // Assert
             Assert.Equal(42, order.AliasedCustomer.ID);
         }
 
+        [Fact]
+        public void ApplyNestedProperty_UsesThePropertyAlias_ForResourceWrapper1()
```

Comment:
I'd advise against naming this test `ApplyNestedProperty_UsesThePropertyAlias_ForResourceWrapper1`. How would someone know how it differs from `ApplyNestedProperty_UsesThePropertyAlias_ForResourceWrapper`?